wandb-scim/examples/custom_roles_example.py

```diff
@@ -0,0 +1,167 @@
+    """
+    Removes permission from a custom role.
+
+    Args:
+        base_url (str): The base URL of the API.
+        custom_role (CustomRole): An instance of the CustomRole class.
+        role_id (str): The ID of the custom role to update.
+        permission_json (str): JSON string representing the permission to remove.
+    """
+    try:
+        # Remove permission from a custom role
+        remove_role_response = custom_role._remove_permission(
+            url=f"{base_url}/Roles/{role_id}",
+            request_payload={
+                "permissionJson": permission_json
+            }
+        )
+        print(remove_role_response)
+    except requests.exceptions.RequestException as e:
+        print(f"Error occurred during API request: {str(e)}")
+
+def update_custom_role(base_url, custom_role, role_id, role_name, role_description, inherited_from):
+    """
+    Updates a custom role.
+
+    Args:
+        base_url (str): The base URL of the API.
+        custom_role (CustomRole): An instance of the CustomRole class.
+        role_id (str): The ID of the custom role to update.
+        role_name (str): The updated name of the custom role.
+        role_description (str): The updated description of the custom role.
+        inherited_from (str): The updated source from which the custom role inherits permissions.
+    """
+    try:
+        # Update a custom role
+        update_custom_role_response = custom_role._update_custom_role(
+            url=f"{base_url}/Roles/{role_id}",
+            request_payload={
+                "roleName": role_name,
+                "roleDescription": role_description,
+                "inheritedFrom": inherited_from
+            }
+        )
+        print(update_custom_role_response)
+    except requests.exceptions.RequestException as e:
+        print(f"Error occurred during API request: {str(e)}")
+
+def delete_custom_role(base_url, custom_role, role_id):
+    """
+    Deletes a custom role.
+
+    Args:
+        base_url (str): The base URL of the API.
+        custom_role (CustomRole): An instance of the CustomRole class.
+        role_id (str): The ID of the custom role to delete.
+    """
+    try:
+        # Delete the custom role
+        delete_role_response = custom_role._delete_custom_role(
+            url=f"{base_url}/Roles/{role_id}"
+        )
+        print(delete_role_response)
+    except requests.exceptions.RequestException as e:
+        print(f"Error occurred during API request: {str(e)}")
+
+if __name__ == "__main__":
+    username = "your_username"
+    api_key = "your_api_key"
+    base_url = "https://example.com/api"  # Base URL of the API
+
+    # Instantiate the CustomRole class with your credentials
+    custom_role = CustomRole(username, api_key)
+    role_id = "abc"  # Replace with actual role ID
+    permission_json = {"name": "project:update"} # Replace with required permissions
+
+    # Test Functions
+    get_all_roles(base_url, custom_role)
```

Comment:
Maybe add a sleep of 5 seconds between each test request, so the invoker user has time to parse / see response of each request before the next one gets invoked.